### PR TITLE
(PC-10293) Move SearchWrapper routes utilities into utils and improvetest readability

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-10293

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [ ] Written **documentation on Notion** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/deeplinks/routing.test.ts [hunk 1]
```diff
@@ -108,7 +108,7 @@ describe('DEEPLINK_TO_SCREEN_CONFIGURATION', () => {
     it('should return Search page when no params are passed', () => {
       const configureScreen = DEEPLINK_TO_SCREEN_CONFIGURATION['recherche']()
       expect(configureScreen.screen).toBe('Search')
-      expect(configureScreen.params).toEqual({
+      expect(configureScreen.params).toStrictEqual({
```
Comment (BenStirrup) [inline src/features/deeplinks/routing.test.ts:111]: Ca va partir

**Review by BenStirrup — APPROVED** [review]

File: src/features/deeplinks/routing.test.ts (hunk 1 above)
Comment (kopax-polyconseil) [reply src/features/deeplinks/routing.test.ts]: undo done

File: src/features/search/utils/tests/sanitizeSearchStateParams.test.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    })
+    expect(sanitizeSearchStateParams({})).toEqual({
+      ...sanitizedSearchStateRequiredDefaults,
+      query: '',
+    })
+  })
+
+  it('should sanitize offerCategories to empty array when not provided', () => {
+    expect(
+      sanitizeSearchStateParams({
```
Comment (antoinewg) [inline src/features/search/utils/tests/sanitizeSearchStateParams.test.ts:25]: cool les tests, mais je pense qu'on peut gagner en lisibilité:

```js
    let { offerCategories } = sanitizeSearchStateParams({ offerCategories: ['CINEMA'] })
    expect(offerCategories).toEqual(['CINEMA'])

    offerCategories = sanitizeSearchStateParams({ offerCategories: undefined }).offerCategories
    expect(offerCategories).toEqual([])
    expect(offerCategories).not.toBeUndefined()
```

File: src/features/search/utils/tests/sanitizeSearchStateParams.test.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    ${`showResults`}       | ${true}
+    ${`timeRange`}         | ${[0, 24]}
+    ${`venueId`}           | ${1337}
+  `(
+    'should sanitize parameter $parameter to "$value" when set and to "undefined" when undefined',
+    ({
+      parameter,
+      value,
+    }: {
+      parameter: string
```
Comment (antoinewg) [inline src/features/search/utils/tests/sanitizeSearchStateParams.test.ts:81]: `keyof SearchState`

et après

```js

      const definedParams = sanitizeSearchStateParams({ [parameter]: value })
      expect(definedParams[parameter]).toEqual(value)

      const undefinedParams = sanitizeSearchStateParams({ [parameter]: undefined })
      expect(undefinedParams[parameter]).toBeUndefined()
```

Comment (kopax-polyconseil) [reply src/features/search/utils/tests/sanitizeSearchStateParams.test.ts:81]: OK

File: src/features/search/utils/tests/sanitizeSearchStateParams.test.ts (hunk 2 above)
Comment (kopax-polyconseil) [reply src/features/search/utils/tests/sanitizeSearchStateParams.test.ts:25]: ok

File: src/features/search/utils/tests/sanitizeSearchStateParams.test.ts (hunk 3 above)
Comment (kopax-polyconseil) [reply src/features/search/utils/tests/sanitizeSearchStateParams.test.ts:81]: https://github.com/pass-culture/pass-culture-app-native/pull/1528

File: src/features/search/utils/tests/sanitizeSearchStateParams.test.ts (hunk 2 above)
Comment (kopax-polyconseil) [reply src/features/search/utils/tests/sanitizeSearchStateParams.test.ts:25]: https://github.com/pass-culture/pass-culture-app-native/pull/1528

**Pull request merged**